remote-wallet/src/trezor.rs

```diff
@@ -0,0 +1,265 @@
+                    .clone();
+                Ok(FirmwareVersion::new(
+                    features.major_version().into(),
+                    features.minor_version().into(),
+                    features.patch_version().into(),
+                ))
+            }
+            _ => Err(RemoteWalletError::NoDeviceFound),
+        }
+    }
+
+    pub fn get_trezor_model(&self) -> Result<String, RemoteWalletError> {
+        match &self.trezor_client {
+            Some(trezor_client) => {
+                let features = trezor_client
+                    .borrow_mut()
+                    .features()
+                    .expect("no features")
+                    .clone();
+                Ok(features.model().to_string())
+            }
+            _ => Err(RemoteWalletError::NoDeviceFound),
+        }
+    }
+
+    pub fn get_trezor_device_id(&self) -> Result<String, RemoteWalletError> {
+        match &self.trezor_client {
+            Some(trezor_client) => {
+                let features = trezor_client
+                    .borrow_mut()
+                    .features()
+                    .expect("no features")
+                    .clone();
+                Ok(features.device_id().to_string())
+            }
+            _ => Err(RemoteWalletError::NoDeviceFound),
+        }
+    }
+}
+
+impl RemoteWallet<Trezor> for TrezorWallet {
+    fn name(&self) -> &str {
+        "Trezor hardware wallet"
+    }
+
+    /// Parse device info and get device base pubkey
+    fn read_device(&mut self, _dev_info: &Trezor) -> Result<RemoteWalletInfo, RemoteWalletError> {
+        unimplemented!();
+    }
+
+    /// Get solana pubkey from a RemoteWallet
+    fn get_pubkey(
+        &self,
+        derivation_path: &DerivationPath,
+        confirm_key: bool,
+    ) -> Result<Pubkey, RemoteWalletError> {
+        let derivation_path_string = format!("{:?}", derivation_path);
+        let derivation_path_str = derivation_path_string.as_str();
+        match &self.trezor_client {
+            Some(trezor_client) => {
+                let mut solana_get_pubkey = SolanaGetPublicKey::new();
+                let address_n = convert_path(
+                    &bip32::DerivationPath::from_str(derivation_path_str)
+                        .expect("Should have properly formatted derivation path for converting"),
+                );
+                solana_get_pubkey.address_n = address_n;
+                solana_get_pubkey.show_display = Some(confirm_key);
+                if confirm_key {
+                    println!("Waiting for your approval on {}", self.name());
+                }
+                let pubkey = handle_interaction(
+                    trezor_client
+                        .borrow_mut()
+                        .call(solana_get_pubkey, Box::new(|_, m: SolanaPublicKey| Ok(m)))?,
+                )?;
+                if confirm_key {
+                    println!("{CHECK_MARK}Approved");
+                }
```

Comment:
I don't think adding a wrapper is necessary. I think it makes sense to only have those two print statements. The first would signify to the user to begin the physical interaction with their Trezor device. As they are moving along their physical interaction feedback is being displayed on the Trezor device. Then, once they have completed their physical interaction on the Trezor device, the second print statement signifies the end of the interaction.

Reply:
Perhaps I wasn't clear enough. Currently the `trezor` and `ledger` behavior for message signing are different, because ledger signing prints these two messages.
Adding a wrapper seems like a nice way to make the behavior consistent.